service-base/src/main/java/org/eclipse/hono/service/AbstractAdapterConfig.java

```diff
@@ -444,15 +446,19 @@ public HonoConnection commandConsumerConnection() {
      * Exposes a factory for creating clients for receiving upstream commands
      * via the AMQP Messaging Network.
      *
+     * @param appContext The Spring application context.
      * @return The factory.
      */
     @Bean
     @Scope("prototype")
-    public ProtocolAdapterCommandConsumerFactory commandConsumerFactory() {
+    public ProtocolAdapterCommandConsumerFactory commandConsumerFactory(final ApplicationContext appContext) {
         // be sure to use a new ClientConfigProperties instance so that its unique container-id
         // is only used for the specific consumer factory instance created here (as adapterInstanceId, which has to be unique)
         final ClientConfigProperties clientConfigProperties = new ClientConfigProperties(commandConsumerFactoryConfig());
         final HonoConnection honoConnection = HonoConnection.newConnection(vertx(), clientConfigProperties);
+        // make sure the tracer instance gets autowired into the honoConnection
+        appContext.getAutowireCapableBeanFactory()
```

Comment:
this should work but wouldn't it be more straight-forward, easier to understand and less Spring dependent to simply add the Tracer as a parameter to `HonoConnectionImpl`'s constructor?

Reply:
yes, but I wanted to keep the change for 1.2.x here as small as possible. (In master, this change isn't needed.) 

In master we can then get rid of autowiring the tracer instance.